app/src/main/java/com/gathering/android/auth/signup/SignUpScreen.kt

```diff
@@ -68,4 +116,53 @@ class SignUpScreen : DialogFragment(), SignUpNavigator {
             R.id.action_signUpScreen_to_verificationScreen, bundle
         )
     }
+
+    @Preview(showBackground = true, device = "spec:parent=pixel_7")
+    @Composable
+    fun SignUpScreenPreview() {
+        LogUpScreen(false, "")
+    }
+
+    @Composable
+    private fun LogUpScreen(isInProgress: Boolean, error: String? = null) {
+        var email by rememberSaveable { mutableStateOf("") }
+        var password by rememberSaveable { mutableStateOf("") }
+        var confirmPassword by rememberSaveable { mutableStateOf("") }
+
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .verticalScroll(rememberScrollState())
+                    .padding(10.dp),
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+
+                GatheringEmailTextField(
+                    value = email,
+                    onValueChange = { email = it },
+                    label = "email"
+                )
+
+                GatheringPasswordTextField(
+                    value = password,
+                    onValueChange = { password = it },
+                    label = "password"
+                )
+
+                GatheringPasswordTextField(
+                    value = confirmPassword,
+                    onValueChange = { confirmPassword = it },
+                    label = "confirm Password"
+                )
+
+                CustomButton(
+                    text = "Sign Up",
+                    onClick = { viewModel.onSignUpButtonClicked(email, password, confirmPassword) },
+                    error = error,
```

Comment:
error shouldnt be a part of button unless its shown inside it. if its a button plus an error maybe its not a button any more. i would suggest to extract error to something like a CustomError composable or something.